# HG changeset patch
# User John Bieling <john@thunderbird.net>
# Date 1710450621 -3600
# Node ID 18c450afb1ab9ef555f90ad303e8940350732388
# Parent  17ac4a0d5d2b3a1ff7dac8c940de6c675b2755a6
Bug 1854138 - Fix onbeforeunload event not preventing content tabs from being closed. r=mkmelin

The return value of browser.permitUnload() was used wrongly.

Differential Revision: https://phabricator.services.mozilla.com/D201825

diff --git a/mail/base/content/specialTabs.js b/mail/base/content/specialTabs.js
--- a/mail/base/content/specialTabs.js
+++ b/mail/base/content/specialTabs.js
@@ -947,9 +947,10 @@ var specialTabs = {
 
       this.lastBrowserId++;
     },
     tryCloseTab(aTab) {
-      return aTab.browser.permitUnload();
+      const { permitUnload } = aTab.browser.permitUnload();
+      return permitUnload;
     },
     persistTab(aTab) {
       if (aTab.browser.currentURI.spec == "about:blank") {
         return null;
